tests/components/humidifier/test_intent.py

```diff
@@ -0,0 +1,208 @@
+
+    result = await hass.helpers.intent.async_handle(
+        "test",
+        intent.INTENT_HUMIDITY,
+        {"name": {"value": "Bedroom humidifier"}, "humidity": {"value": "50"}},
+    )
+    await hass.async_block_till_done()
+
+    assert result.speech["plain"]["speech"] == "The bedroom humidifier is set to 50%"
+
+    assert len(turn_on_calls) == 0
+    assert len(humidity_calls) == 1
+    call = humidity_calls[0]
+    assert call.domain == DOMAIN
+    assert call.service == SERVICE_SET_HUMIDITY
+    assert call.data.get(ATTR_ENTITY_ID) == "humidifier.bedroom_humidifier"
+    assert call.data.get(ATTR_HUMIDITY) == 50
+
+
+async def test_intent_set_humidity_and_turn_on(hass):
+    """Test the set humidity intent for turned off humidifier."""
+    hass.states.async_set(
+        "humidifier.bedroom_humidifier", STATE_OFF, {ATTR_HUMIDITY: 40}
+    )
+    humidity_calls = async_mock_service(hass, DOMAIN, SERVICE_SET_HUMIDITY)
+    turn_on_calls = async_mock_service(hass, DOMAIN, SERVICE_TURN_ON)
+    await intent.async_setup_intents(hass)
+
+    result = await hass.helpers.intent.async_handle(
+        "test",
+        intent.INTENT_HUMIDITY,
+        {"name": {"value": "Bedroom humidifier"}, "humidity": {"value": "50"}},
+    )
+    await hass.async_block_till_done()
+
+    assert (
+        result.speech["plain"]["speech"]
+        == "Turned bedroom humidifier on and set humidity to 50%"
+    )
+
+    assert len(turn_on_calls) == 1
+    call = turn_on_calls[0]
+    assert call.domain == DOMAIN
+    assert call.service == SERVICE_TURN_ON
+    assert call.data.get(ATTR_ENTITY_ID) == "humidifier.bedroom_humidifier"
+    assert len(humidity_calls) == 1
+    call = humidity_calls[0]
+    assert call.domain == DOMAIN
+    assert call.service == SERVICE_SET_HUMIDITY
+    assert call.data.get(ATTR_ENTITY_ID) == "humidifier.bedroom_humidifier"
+    assert call.data.get(ATTR_HUMIDITY) == 50
+
+
+async def test_intent_set_mode(hass):
+    """Test the set mode intent."""
+    hass.states.async_set(
+        "humidifier.bedroom_humidifier",
+        STATE_ON,
+        {
+            ATTR_HUMIDITY: 40,
+            ATTR_SUPPORTED_FEATURES: 1,
+            ATTR_AVAILABLE_MODES: ["home", "away"],
+            ATTR_MODE: "home",
+        },
+    )
+    mode_calls = async_mock_service(hass, DOMAIN, SERVICE_SET_MODE)
+    turn_on_calls = async_mock_service(hass, DOMAIN, SERVICE_TURN_ON)
+    await intent.async_setup_intents(hass)
+
+    result = await hass.helpers.intent.async_handle(
+        "test",
+        intent.INTENT_MODE,
+        {"name": {"value": "Bedroom humidifier"}, "mode": {"value": "away"}},
+    )
+    await hass.async_block_till_done()
+
+    assert (
+        result.speech["plain"]["speech"]
+        == "The mode for bedroom humidifier is set to away"
+    )
+
+    assert len(turn_on_calls) == 0
+    assert len(mode_calls) == 1
+    call = mode_calls[0]
+    assert call.domain == DOMAIN
+    assert call.service == SERVICE_SET_MODE
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)